cmd/osv-scanner/main.go

```diff
@@ -25,38 +26,39 @@ const osvScannerConfigName = "osv-scanner.toml"
 //   - Any lockfiles with scanLockfile
 //   - Any SBOM files with scanSBOMFile
 //   - Any git repositories with scanGit
-func scanDir(query *osv.BatchedQuery, dir string, skipGit bool, recursive bool) error {
+func scanDir(r *output.Reporter, query *osv.BatchedQuery, dir string, skipGit bool, recursive bool) error {
 	root := true
 	return filepath.WalkDir(dir, func(path string, info os.DirEntry, err error) error {
 		if err != nil {
-			log.Printf("Failed to walk %s: %v", path, err)
+			r.PrintText(fmt.Sprintf("Failed to walk %s: %v\n", path, err))
```

Comment:
`f.PrintText(fmt.Sprintf(` is common enough to just have `f.PrintText()` be `f.PrintTextf()` with a function signature same as `fmt.Sprintf()`, reducing the need to always put `fmt.Sprintf()` in. Same with `PrintError`

Reply:
I've deliberately not done that because [Golang doesn't (yet) support custom string formatting functions](https://youtrack.jetbrains.com/issue/GO-5841/Allow-format-message-inspection-customization) and I've found it's more useful having the IDE helper vs the couple of extra characters you save (+ because this is Go it doesn't impact the compiled binary size)

Reply:
Huh, that's fair enough, thanks for pointing that out.